packages/eslint-plugin-template/src/rules/no-duplicate-attributes.ts

```diff
@@ -0,0 +1,158 @@
+import {
+  BindingType,
+  TmplAstBoundAttribute,
+  TmplAstBoundEvent,
+  TmplAstElement,
+  TmplAstTextAttribute,
+  ParsedEventType,
+} from '@angular/compiler';
+import {
+  createESLintRule,
+  getTemplateParserServices,
+} from '../utils/create-eslint-rule';
+
+type Options = [
+  {
+    allowTwoWayDataBinding?: boolean;
+  },
+];
+export type MessageIds = 'noDuplicateAttributes';
+export const RULE_NAME = 'no-duplicate-attributes';
+
+const defaultOptions = {
+  allowTwoWayDataBinding: true,
+};
+
+export default createESLintRule<Options, MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'problem',
+    docs: {
+      description:
+        'Ensures that there are no duplicate input properties or output event listeners',
+      category: 'Possible Errors',
+      recommended: false,
+    },
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          allowTwoWayDataBinding: {
+            type: 'boolean',
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+    messages: {
+      noDuplicateAttributes: 'Duplicate attribute "{{attributeName}}"',
+    },
+  },
+  defaultOptions: [defaultOptions],
```

Comment:
nit: we generally prefer to inline these options:

```suggestion
  defaultOptions: [{ allowTwoWayDataBinding: true }],
```